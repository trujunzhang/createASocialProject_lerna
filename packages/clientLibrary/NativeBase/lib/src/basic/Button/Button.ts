import * as React from 'react'

import { ReactNative, RnViewStyleProp, RnTextStyleProp, BsStyle } from '@appUtils/rnCommonProps'

/**
 * see Widget Button.js
 */
export interface ButtonProps extends ReactNative.TouchableOpacityProps, BsStyle {
  /**
   * Defines button style
   */
  style?: RnViewStyleProp | Array<RnViewStyleProp>
  /**
   * Defines button text style
   */
  textStyle?: RnTextStyleProp
  // onPress?: () => void
  /**
   * Block level button
   */
  block?: boolean
  /**
   * Vertical button
   */
  vertical?: boolean
  /**
   * Button with badges
   */
  badge?: boolean
  //primary?: boolean,
  /**
   * Gives you effect of Icon-buttons.
   * To have button with transparent background, include this prop.
   */
  transparent?: boolean
  //success?: boolean,
  //danger?: boolean,
  // warning?: boolean,
  //info?: boolean,
  color?: string
  /**
   * Applies outline button style.
   */
  bordered?: boolean
  /**
   * Renders button with slightly round shaped edges.
   */
  rounded?: boolean
  /**
   * For large size button
   */
  large?: boolean
  /**
   * For small size button
   */
  small?: boolean
  /**
   * Used for Icon alignment.
   * Aligns icon to the left in button.
   * By default, icons are aligned to the left in button.
   */
  iconLeft?: boolean
  /**
   * Used for Icon alignment.
   * Aligns icon to the right in button.
   */
  iconRight?: boolean
  /**
   * Used to enable/disable icons
   * Icons align in the center of the button.
   */
  icon?: boolean
  /**
   * Disables onPress option for button
   */
  disabled?: boolean
  active?: boolean
  inputButton?: boolean
  full?: boolean
  light?: boolean
  dark?: boolean
  /**
   * [android] colored ripple effect
   */
  androidRippleColor?: string
  /**
   * Segments
   */
  first?: boolean
  last?: boolean
  /**
   * Adds necessary padding when Text button defined in Left / Right of Header (iOS)
   */
  hasText?: boolean
  // Others
  activeOpacity?: any
  onlyCenter?: boolean
  facebook?: boolean
  twitter?: boolean
  svgLeft?: boolean
}
