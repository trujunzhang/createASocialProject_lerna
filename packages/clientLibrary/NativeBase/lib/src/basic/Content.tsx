import * as React from 'react'
import * as ReactNative from 'react-native'
import * as PropTypes from 'prop-types'
import { KeyboardAwareScrollView } from '@app/react-native-keyboard-aware-scroll-view'
import { connectStyle } from '@app/native-base-shoutem-theme'
import mapPropsToStyleNames from '../utils/mapPropsToStyleNames'
import { platformVariables as variable } from '@app/native-base-variables'
import { isIphoneX } from '@app/react-native-iphone-x-helper'

export interface IContentProps {
  style?: ReactNative.ViewStyle | Array<ReactNative.ViewStyle> | any
  /**
     * The theme prop can be applied to any component of NativeBase.
     */
  refreshing?: boolean
  refreshControl?: object | any
  theme?: Object
  padder?: boolean
  disableKBDismissScroll?: boolean | any
  enableResetScrollToCoords?: boolean
  contentOffset?: Object | any
  scrollEnabled?: boolean
  contentContainerStyle?: ReactNative.ViewStyle | Array<ReactNative.ViewStyle>
  keyboardShouldPersistTaps?: string | any
  keyboardDismissMode?: string | any
}
class Content extends React.Component<IContentProps, any> {
  private _root: any
  private _scrollview: any
  static contextTypes = {
    theme: PropTypes.object
  }
  constructor(props) {
    super(props)
    this.state = {
      orientation: 'portrait'
    }
  }
  layoutChange(val) {
    let maxComp = Math.max(variable.deviceWidth, variable.deviceHeight)
    if (val.width >= maxComp) this.setState({ orientation: 'landscape' })
    else {
      this.setState({ orientation: 'portrait' })
    }
  }

  calculateLeft(mode, inSet) {
    let inset: any = null
    if (inSet !== undefined) {
      inset = inSet
    } else {
      inset = variable.Inset
    }
    const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape
    let leftPadder = null
    if (this.props.style[1] !== undefined) {
      if (
        this.props.style[1].padding !== undefined ||
        this.props.style[1].paddingLeft !== undefined
      ) {
        leftPadder =
          (this.props.style[1].paddingLeft
            ? this.props.style[1].paddingLeft
            : this.props.style[1].padding) + InsetValues.leftInset
      }
    } else if (
      this.props.style.padding !== undefined ||
      this.props.style.paddingLeft !== undefined
    ) {
      leftPadder =
        (this.props.style.paddingLeft ? this.props.style.paddingLeft : this.props.style.padding) +
        InsetValues.leftInset
    } else {
      leftPadder = InsetValues.leftInset
    }
    return leftPadder
  }
  calculateRight(mode, inSet) {
    let inset: any = null
    if (inSet !== undefined) {
      inset = inSet
    } else {
      inset = variable.Inset
    }
    const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape
    let rightPadder = null
    if (this.props.style[1] !== undefined) {
      if (
        this.props.style[1].padding !== undefined ||
        this.props.style[1].paddingRight !== undefined
      ) {
        rightPadder =
          (this.props.style[1].paddingRight
            ? this.props.style[1].paddingRight
            : this.props.style[1].padding) + InsetValues.rightInset
      }
    } else if (
      this.props.style.padding !== undefined ||
      this.props.style.paddingRight !== undefined
    ) {
      rightPadder =
        (this.props.style.paddingRight ? this.props.style.paddingRight : this.props.style.padding) +
        InsetValues.rightInset
    } else {
      rightPadder = InsetValues.rightInset
    }
    return rightPadder
  }
  render() {
    const variables = this.context.theme
      ? this.context.theme['@@shoutem.theme/themeStyle'].variables
      : variable
    return isIphoneX() ? (
      <KeyboardAwareScrollView
        automaticallyAdjustContentInsets={false}
        resetScrollToCoords={(this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }) as any}
        keyboardShouldPersistTaps={
          this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled'
        }
        ref={(c) => {
          this._scrollview = c
          this._root = c
        }}
        {...this.props}
        onLayout={(e) => this.layoutChange(e.nativeEvent.layout)}
        style={[
          this.props.style,
          {
            paddingLeft: this.calculateLeft(this.state.orientation, variables.Inset)
          }
        ]}
        contentContainerStyle={[
          { padding: this.props.padder ? variables.contentPadding : undefined },
          this.props.contentContainerStyle
        ]}>
        {this.props.children}
      </KeyboardAwareScrollView>
    ) : (
        <KeyboardAwareScrollView
          automaticallyAdjustContentInsets={false}
          resetScrollToCoords={(this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }) as any}
          keyboardShouldPersistTaps={
            this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled'
          }
          ref={(c) => {
            this._scrollview = c
            this._root = c
          }}
          {...this.props}
          contentContainerStyle={[
            { padding: this.props.padder ? variables.contentPadding : undefined },
            this.props.contentContainerStyle
          ]}>
          {this.props.children}
        </KeyboardAwareScrollView>
      )
  }
}

// Content.propTypes = {
//     style: PropTypes.oneOfType([
//         PropTypes.object,
//         PropTypes.number,
//         PropTypes.array
//     ]),
//     padder: PropTypes.bool,
//     disableKBDismissScroll: PropTypes.bool,
//     enableResetScrollToCoords: PropTypes.bool,
//     keyboardShouldPersistTaps: PropTypes.string
// };

const StyledContent = connectStyle('NativeBase.Content', {}, mapPropsToStyleNames)(Content)

export { StyledContent as Content }
