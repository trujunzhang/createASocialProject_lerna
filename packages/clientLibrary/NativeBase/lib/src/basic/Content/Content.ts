import * as React from 'react'
import { ReactNative, RnTextStyleProp, RnViewStyleProp } from '@appUtils/rnCommonProps'

/**
 * see Widget Content.js
 */
export interface ContentProps {
  /**
   * The theme prop can be applied to any component of NativeBase.
   */
  refreshing?: boolean
  refreshControl?: object
  theme?: Object
  padder?: boolean
  disableKBDismissScroll?: boolean
  enableResetScrollToCoords?: boolean
  contentOffset?: Object
  scrollEnabled?: boolean
  style?: RnViewStyleProp | Array<RnViewStyleProp>
  contentContainerStyle?: RnViewStyleProp | Array<RnViewStyleProp>
  keyboardShouldPersistTaps?: string
  keyboardDismissMode?: string
}
