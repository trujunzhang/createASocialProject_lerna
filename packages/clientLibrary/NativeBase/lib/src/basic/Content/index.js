import * as React from 'react';
import * as PropTypes from 'prop-types';
// TODO:DJZHANG(NB)
import { getKeyboardAwareScrollView } from '@appUtils/cacheHelper';
import { isIphoneX } from 'react-primitives';
// import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import { connectStyle } from 'native-base-shoutem-theme';
import { platformVariables as variable } from '@appVariables/theme/variables/platform'; // variables // [variable]
import { mapPropsToStyleNames } from '@app/tools';
class Content extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            orientation: 'portrait'
        };
    }
    layoutChange(val) {
        let maxComp = Math.max(variable.deviceWidth, variable.deviceHeight);
        if (val.width >= maxComp)
            this.setState({ orientation: 'landscape' });
        else {
            this.setState({ orientation: 'portrait' });
        }
    }
    calculateLeft(mode, inSet) {
        let inset = null;
        if (inSet !== undefined) {
            inset = inSet;
        }
        else {
            inset = variable.Inset;
        }
        const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape;
        let leftPadder = null;
        if (this.props.style[1] !== undefined) {
            if (this.props.style[1].padding !== undefined ||
                this.props.style[1].paddingLeft !== undefined) {
                leftPadder =
                    (this.props.style[1].paddingLeft
                        ? this.props.style[1].paddingLeft
                        : this.props.style[1].padding) + InsetValues.leftInset;
            }
        }
        else if (this.props.style.padding !== undefined ||
            this.props.style.paddingLeft !== undefined) {
            leftPadder =
                (this.props.style.paddingLeft ? this.props.style.paddingLeft : this.props.style.padding) +
                    InsetValues.leftInset;
        }
        else {
            leftPadder = InsetValues.leftInset;
        }
        return leftPadder;
    }
    calculateRight(mode, inSet) {
        let inset = null;
        if (inSet !== undefined) {
            inset = inSet;
        }
        else {
            inset = variable.Inset;
        }
        const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape;
        let rightPadder = null;
        if (this.props.style[1] !== undefined) {
            if (this.props.style[1].padding !== undefined ||
                this.props.style[1].paddingRight !== undefined) {
                rightPadder =
                    (this.props.style[1].paddingRight
                        ? this.props.style[1].paddingRight
                        : this.props.style[1].padding) + InsetValues.rightInset;
            }
        }
        else if (this.props.style.padding !== undefined ||
            this.props.style.paddingRight !== undefined) {
            rightPadder =
                (this.props.style.paddingRight ? this.props.style.paddingRight : this.props.style.padding) +
                    InsetValues.rightInset;
        }
        else {
            rightPadder = InsetValues.rightInset;
        }
        return rightPadder;
    }
    render() {
        const variables = this.context.theme
            ? this.context.theme['@@shoutem.theme/themeStyle'].variables
            : variable;
        const KeyboardAwareScrollView = getKeyboardAwareScrollView();
        return isIphoneX() ? (React.createElement(KeyboardAwareScrollView, Object.assign({ name: 'keyboardAwareScrollView', automaticallyAdjustContentInsets: false, resetScrollToCoords: this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }, keyboardShouldPersistTaps: this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled', ref: (c) => {
                this._scrollview = c;
                this._root = c;
            } }, this.props, { onLayout: (e) => this.layoutChange(e.nativeEvent.layout), style: [
                this.props.style,
                {
                    paddingLeft: this.calculateLeft(this.state.orientation, variables.Inset)
                }
            ], contentContainerStyle: [
                { padding: this.props.padder ? variables.contentPadding : undefined },
                this.props.contentContainerStyle
            ] }), this.props.children)) : (React.createElement(KeyboardAwareScrollView, Object.assign({ name: 'keyboardAwareScrollView', automaticallyAdjustContentInsets: false, resetScrollToCoords: this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }, keyboardShouldPersistTaps: this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled', ref: (c) => {
                this._scrollview = c;
                this._root = c;
            } }, this.props, { contentContainerStyle: [
                { padding: this.props.padder ? variables.contentPadding : undefined },
                this.props.contentContainerStyle
            ] }), this.props.children));
    }
}
Content.contextTypes = {
    theme: PropTypes.object
};
// Content.propTypes = {
//   style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
//   padder: PropTypes.bool,
//   disableKBDismissScroll: PropTypes.bool,
//   enableResetScrollToCoords: PropTypes.bool,
//   keyboardShouldPersistTaps: PropTypes.string
// }
const StyledContent = connectStyle('NativeBase.Content', {}, mapPropsToStyleNames)(Content);
const TabContent = StyledContent;
export { StyledContent as Content, TabContent };
//# sourceMappingURL=index.js.map