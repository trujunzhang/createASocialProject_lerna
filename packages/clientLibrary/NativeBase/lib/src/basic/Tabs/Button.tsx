import * as React from 'react'
// const React = require('react')
// const ReactNative = require('react-native')

import { TouchableOpacity, View } from 'react-primitives'
export const Button = (props) => {
  return <TouchableOpacity {...props}>{props.children}</TouchableOpacity>
}

// module.exports = Button;
//export default Button;
