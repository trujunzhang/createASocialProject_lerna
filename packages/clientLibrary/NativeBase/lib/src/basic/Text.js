import * as React from 'react'
import PropTypes from "prop-types";
import { Text as RNText } from "react-primitives";
import { LodashUtils as _ } from '@app/tools'

import { connectStyle } from "@app/native-base-shoutem-theme";
import mapPropsToStyleNames from "../utils/mapPropsToStyleNames";

class Text extends React.Component {
  render() {
    const {
      uppercase,
      children,
    } = this.props;

    let text;
    if (uppercase) {
      text = React.Children.map(children, child => {
        if (_.isString(child)) {
          return _.toUpper(child);
        } else {
          return child;
        }
      })
    } else {
      text = children;
    }

    return (
      <RNText ref={c => (this._root = c)} {...this.props}>
        {text}
      </RNText>
    );
  }
}

Text.propTypes = {
  ...RNText.propTypes,
  uppercase: PropTypes.bool,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
};

Text.defaultProps = {
  uppercase: false,
};

const StyledText = connectStyle("NativeBase.Text", {}, mapPropsToStyleNames)(Text);

export { StyledText as Text };
