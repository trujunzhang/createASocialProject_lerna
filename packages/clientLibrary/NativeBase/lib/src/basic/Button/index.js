import * as React from 'react';
import * as PropTypes from 'prop-types';
import { TouchableOpacity, Platform, View, TouchableNativeFeedback, StyleSheet } from 'react-primitives';
import { connectStyle } from 'native-base-shoutem-theme';
import { platformVariables as variable } from '@appVariables/theme/variables/platform'; // variables // [variable]
import { Text } from '@appBasic/Text';
import computeProps from '@appUtils/computeProps';
import { mapPropsToStyleNames } from '@app/tools';
class Button extends React.Component {
    getInitialStyle() {
        return {
            borderedBtn: {
                borderWidth: this.props.bordered ? 1 : undefined,
                borderRadius: this.props.rounded && this.props.bordered ? variable.borderRadiusLarge : 2
            }
        };
    }
    prepareRootProps() {
        const defaultProps = {
            style: this.getInitialStyle().borderedBtn
        };
        if (Array.isArray(this.props.style)) {
            const flattenedStyle = this.props.style.reduce((accumulator, currentValue) => accumulator.concat(currentValue), []);
            return computeProps({ ...this.props, style: flattenedStyle }, defaultProps);
        }
        return computeProps(this.props, defaultProps);
    }
    render() {
        const variables = this.context.theme
            ? this.context.theme['@@shoutem.theme/themeStyle'].variables
            : variable;
        const children = Platform.OS === 'ios'
            ? this.props.children
            : React.Children.map(this.props.children, (child) => child && child.type === Text
                ? React.cloneElement(child, {
                    uppercase: variables.btnUppercaseAndroidText,
                    ...child.props
                })
                : child);
        if (Platform.OS === 'ios' ||
            Platform.OS === 'web' ||
            variables.androidRipple === false ||
            Platform['Version'] < 21) {
            return (React.createElement(TouchableOpacity, Object.assign({}, this.prepareRootProps(), { ref: (c) => (this._root = c), activeOpacity: this.props.activeOpacity > 0 ? this.props.activeOpacity : 0.5 }), children));
        }
        else {
            if (this.props.rounded) {
                let buttonStyle = { ...this.prepareRootProps().style };
                let buttonFlex = this.props.full || this.props.block ? 1 : buttonStyle.flex;
                return (React.createElement(View, { style: [
                        { maxHeight: buttonStyle.height },
                        buttonStyle,
                        { paddingTop: undefined, paddingBottom: undefined }
                    ] },
                    React.createElement(TouchableNativeFeedback, Object.assign({ ref: (c) => (this._root = c), background: this.props.androidRippleColor
                            ? TouchableNativeFeedback.Ripple(this.props.androidRippleColor, true)
                            : TouchableNativeFeedback.Ripple(variables.androidRippleColor, true) }, this.prepareRootProps()),
                        React.createElement(View, { style: [
                                styles.childContainer,
                                {
                                    paddingTop: buttonStyle.paddingTop,
                                    paddingBottom: buttonStyle.paddingBottom,
                                    height: buttonStyle.height,
                                    flexGrow: buttonFlex
                                }
                            ] }, children))));
            }
            else {
                return (React.createElement(TouchableNativeFeedback, Object.assign({ ref: (c) => (this._root = c), onPress: this.props.onPress, background: this.props.transparent
                        ? TouchableNativeFeedback.Ripple('transparent')
                        : TouchableNativeFeedback.Ripple(variables.androidRippleColor, false) }, this.prepareRootProps()),
                    React.createElement(View, Object.assign({}, this.prepareRootProps()), children)));
            }
        }
    }
}
Button.contextTypes = {
    theme: PropTypes.object
};
// Button.propTypes = {
//   ...TouchableOpacity.propTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ]),
//   block: PropTypes.bool,
//   primary: PropTypes.bool,
//   transparent: PropTypes.bool,
//   success: PropTypes.bool,
//   danger: PropTypes.bool,
//   warning: PropTypes.bool,
//   info: PropTypes.bool,
//   bordered: PropTypes.bool,
//   disabled: PropTypes.bool,
//   rounded: PropTypes.bool,
//   large: PropTypes.bool,
//   small: PropTypes.bool,
//   active: PropTypes.bool
// };
const styles = StyleSheet.create({
    childContainer: {
        flexShrink: 1,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center'
    }
});
const StyledButton = connectStyle('NativeBase.Button', {}, mapPropsToStyleNames)(Button);
export { StyledButton as Button };
//# sourceMappingURL=index.js.map