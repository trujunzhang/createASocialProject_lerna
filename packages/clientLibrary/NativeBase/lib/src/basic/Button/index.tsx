import * as React from 'react'
import * as PropTypes from 'prop-types'
import {
  TouchableOpacity,
  Platform,
  View,
  TouchableNativeFeedback,
  StyleSheet
} from 'react-primitives'
import { connectStyle } from 'native-base-shoutem-theme'
import { IThemeVariables } from '@app/tools' // typings
import { platformVariables as variable } from '@appVariables/theme/variables/platform' // variables // [variable]

import { Text } from '@appBasic/Text'
import computeProps from '@appUtils/computeProps'

import { mapPropsToStyleNames } from '@app/tools'

import { ButtonProps } from './Button'

class Button extends React.Component<ButtonProps, any> {
  static contextTypes = {
    theme: PropTypes.object
  }

  getInitialStyle() {
    return {
      borderedBtn: {
        borderWidth: this.props.bordered ? 1 : undefined,
        borderRadius: this.props.rounded && this.props.bordered ? variable.borderRadiusLarge : 2
      }
    }
  }

  // _root: React$Element<TouchableOpacity | TouchableNativeFeedback>;
  _root: any

  prepareRootProps() {
    const defaultProps = {
      style: this.getInitialStyle().borderedBtn
    }

    if (Array.isArray(this.props.style)) {
      const flattenedStyle = this.props.style.reduce(
        (accumulator: any, currentValue) => accumulator.concat(currentValue),
        []
      )
      return computeProps({ ...this.props, style: flattenedStyle }, defaultProps)
    }

    return computeProps(this.props, defaultProps)
  }
  render() {
    const variables: IThemeVariables = this.context.theme
      ? this.context.theme['@@shoutem.theme/themeStyle'].variables
      : variable
    const children =
      Platform.OS === 'ios'
        ? this.props.children
        : React.Children.map(this.props.children, (child: any) =>
            child && child.type === Text
              ? React.cloneElement(child, {
                  uppercase: variables.btnUppercaseAndroidText,
                  ...child.props
                })
              : child
          )
    if (
      Platform.OS === 'ios' ||
      Platform.OS === 'web' ||
      variables.androidRipple === false ||
      Platform['Version'] < 21
    ) {
      return (
        <TouchableOpacity
          {...this.prepareRootProps()}
          ref={(c) => (this._root = c)}
          activeOpacity={this.props.activeOpacity > 0 ? this.props.activeOpacity : 0.5}>
          {children}
        </TouchableOpacity>
      )
    } else {
      if (this.props.rounded) {
        let buttonStyle = { ...this.prepareRootProps().style }
        let buttonFlex = this.props.full || this.props.block ? 1 : buttonStyle.flex
        return (
          <View
            style={[
              { maxHeight: buttonStyle.height },
              buttonStyle,
              { paddingTop: undefined, paddingBottom: undefined }
            ]}>
            <TouchableNativeFeedback
              ref={(c) => (this._root = c)}
              background={
                this.props.androidRippleColor
                  ? TouchableNativeFeedback.Ripple(this.props.androidRippleColor, true)
                  : TouchableNativeFeedback.Ripple(variables.androidRippleColor, true)
              }
              {...this.prepareRootProps()}>
              <View
                style={[
                  styles.childContainer,
                  {
                    paddingTop: buttonStyle.paddingTop,
                    paddingBottom: buttonStyle.paddingBottom,
                    height: buttonStyle.height,
                    flexGrow: buttonFlex
                  }
                ]}>
                {children}
              </View>
            </TouchableNativeFeedback>
          </View>
        )
      } else {
        return (
          <TouchableNativeFeedback
            ref={(c) => (this._root = c)}
            onPress={this.props.onPress}
            background={
              this.props.transparent
                ? TouchableNativeFeedback.Ripple('transparent')
                : TouchableNativeFeedback.Ripple(variables.androidRippleColor, false)
            }
            {...this.prepareRootProps()}>
            <View {...this.prepareRootProps()}>{children}</View>
          </TouchableNativeFeedback>
        )
      }
    }
  }
}

// Button.propTypes = {
//   ...TouchableOpacity.propTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ]),
//   block: PropTypes.bool,
//   primary: PropTypes.bool,
//   transparent: PropTypes.bool,
//   success: PropTypes.bool,
//   danger: PropTypes.bool,
//   warning: PropTypes.bool,
//   info: PropTypes.bool,
//   bordered: PropTypes.bool,
//   disabled: PropTypes.bool,
//   rounded: PropTypes.bool,
//   large: PropTypes.bool,
//   small: PropTypes.bool,
//   active: PropTypes.bool
// };

const styles = StyleSheet.create({
  childContainer: {
    flexShrink: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  }
})

const StyledButton = connectStyle('NativeBase.Button', {}, mapPropsToStyleNames)(Button)
export { StyledButton as Button }
