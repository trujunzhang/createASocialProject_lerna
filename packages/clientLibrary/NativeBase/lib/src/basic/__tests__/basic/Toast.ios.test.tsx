import 'react-primitives'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Content } from '../../Content'
import { Button } from '../../Button'
import { Toast } from '../../Toast'
import { Text } from '../../Text'

// Note: test renderer must be required after react-native.
// const cover = require("../assets/drawer-cover.png");

// jest.mock("Platform", () => {
//   const Platform = require.requireActual("Platform");
//   Platform.OS = "ios";
//  return Platform;
// });

it('renders default Toast', () => {
  const tree = renderer
    .create(
      <Button
        onPress={() =>
          Toast.show({
            text: 'Wrong password!',
            buttonText: 'Okay'
          })
        }>
        <Text>Toast</Text>
      </Button>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Toast button style', () => {
  const tree = renderer
    .create(
      <Button
        onPress={() =>
          Toast.show({
            text: 'Wrong password!',
            buttonText: 'Okay',
            buttonTextStyle: { color: '#008000' },
            buttonStyle: { backgroundColor: '#5cb85c' }
          })
        }>
        <Text>Toast</Text>
      </Button>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Toast with duration', () => {
  const tree = renderer
    .create(
      <Button
        onPress={() =>
          Toast.show({
            text: 'Wrong password!',
            buttonText: 'Okay',
            duration: 3000
          })
        }>
        <Text>Toast</Text>
      </Button>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Toast position', () => {
  const tree = renderer
    .create(
      <Content padder>
        <Button
          onPress={() =>
            Toast.show({
              text: 'Wrong password!',
              buttonText: 'Okay',
              position: 'top'
            })
          }>
          <Text>Top Toast</Text>
        </Button>
        <Button
          onPress={() =>
            Toast.show({
              text: 'Wrong password!',
              buttonText: 'Okay',
              position: 'bottom'
            })
          }>
          <Text>Bottom Toast</Text>
        </Button>
      </Content>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Toast Text style', () => {
  const tree = renderer
    .create(
      <Button
        onPress={() =>
          Toast.show({
            text: 'Wrong password!',
            textStyle: { color: 'yellow' },
            buttonText: 'Okay'
          })
        }>
        <Text>Toast</Text>
      </Button>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Toast Type', () => {
  const tree = renderer
    .create(
      <Content padder>
        <Button
          onPress={() =>
            Toast.show({
              text: 'Wrong password!',
              buttonText: 'Okay'
            })
          }>
          <Text>Default Toast</Text>
        </Button>
        <Button
          success
          onPress={() =>
            Toast.show({
              text: 'Wrong password!',
              buttonText: 'Okay',
              type: 'success'
            })
          }>
          <Text>Success Toast</Text>
        </Button>
        <Button
          warning
          onPress={() =>
            Toast.show({
              text: 'Wrong password!',
              buttonText: 'Okay',
              type: 'warning'
            })
          }>
          <Text>Warning Toast</Text>
        </Button>
        <Button
          danger
          onPress={() =>
            Toast.show({
              text: 'Wrong password!',
              buttonText: 'Okay',
              type: 'danger'
            })
          }>
          <Text>Danger Toast</Text>
        </Button>
      </Content>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
