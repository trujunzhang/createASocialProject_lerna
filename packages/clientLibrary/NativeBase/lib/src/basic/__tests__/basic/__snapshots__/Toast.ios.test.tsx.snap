// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Toast Text style 1`] = `
<view
  name="View"
  style={
    Object {
      "borderBottomLeftRadius": 2,
      "borderBottomRightRadius": 2,
      "borderTopLeftRadius": 2,
      "borderTopRightRadius": 2,
    }
  }
>
  <view
    name="View"
    style={Object {}}
  >
    <text
      style={Object {}}
    >
      Toast
    </text>
  </view>
</view>
`;

exports[`renders Toast Type 1`] = `
<view
  name="keyboardAwareScrollView"
  style={Object {}}
>
  <view
    name="View"
    style={
      Object {
        "borderBottomLeftRadius": 2,
        "borderBottomRightRadius": 2,
        "borderTopLeftRadius": 2,
        "borderTopRightRadius": 2,
      }
    }
  >
    <view
      name="View"
      style={Object {}}
    >
      <text
        style={Object {}}
      >
        Default Toast
      </text>
    </view>
  </view>
  <view
    name="View"
    style={
      Object {
        "borderBottomLeftRadius": 2,
        "borderBottomRightRadius": 2,
        "borderTopLeftRadius": 2,
        "borderTopRightRadius": 2,
      }
    }
  >
    <view
      name="View"
      style={Object {}}
    >
      <text
        style={Object {}}
      >
        Success Toast
      </text>
    </view>
  </view>
  <view
    name="View"
    style={
      Object {
        "borderBottomLeftRadius": 2,
        "borderBottomRightRadius": 2,
        "borderTopLeftRadius": 2,
        "borderTopRightRadius": 2,
      }
    }
  >
    <view
      name="View"
      style={Object {}}
    >
      <text
        style={Object {}}
      >
        Warning Toast
      </text>
    </view>
  </view>
  <view
    name="View"
    style={
      Object {
        "borderBottomLeftRadius": 2,
        "borderBottomRightRadius": 2,
        "borderTopLeftRadius": 2,
        "borderTopRightRadius": 2,
      }
    }
  >
    <view
      name="View"
      style={Object {}}
    >
      <text
        style={Object {}}
      >
        Danger Toast
      </text>
    </view>
  </view>
</view>
`;

exports[`renders Toast button style 1`] = `
<view
  name="View"
  style={
    Object {
      "borderBottomLeftRadius": 2,
      "borderBottomRightRadius": 2,
      "borderTopLeftRadius": 2,
      "borderTopRightRadius": 2,
    }
  }
>
  <view
    name="View"
    style={Object {}}
  >
    <text
      style={Object {}}
    >
      Toast
    </text>
  </view>
</view>
`;

exports[`renders Toast position 1`] = `
<view
  name="keyboardAwareScrollView"
  style={Object {}}
>
  <view
    name="View"
    style={
      Object {
        "borderBottomLeftRadius": 2,
        "borderBottomRightRadius": 2,
        "borderTopLeftRadius": 2,
        "borderTopRightRadius": 2,
      }
    }
  >
    <view
      name="View"
      style={Object {}}
    >
      <text
        style={Object {}}
      >
        Top Toast
      </text>
    </view>
  </view>
  <view
    name="View"
    style={
      Object {
        "borderBottomLeftRadius": 2,
        "borderBottomRightRadius": 2,
        "borderTopLeftRadius": 2,
        "borderTopRightRadius": 2,
      }
    }
  >
    <view
      name="View"
      style={Object {}}
    >
      <text
        style={Object {}}
      >
        Bottom Toast
      </text>
    </view>
  </view>
</view>
`;

exports[`renders Toast with duration 1`] = `
<view
  name="View"
  style={
    Object {
      "borderBottomLeftRadius": 2,
      "borderBottomRightRadius": 2,
      "borderTopLeftRadius": 2,
      "borderTopRightRadius": 2,
    }
  }
>
  <view
    name="View"
    style={Object {}}
  >
    <text
      style={Object {}}
    >
      Toast
    </text>
  </view>
</view>
`;

exports[`renders default Toast 1`] = `
<view
  name="View"
  style={
    Object {
      "borderBottomLeftRadius": 2,
      "borderBottomRightRadius": 2,
      "borderTopLeftRadius": 2,
      "borderTopRightRadius": 2,
    }
  }
>
  <view
    name="View"
    style={Object {}}
  >
    <text
      style={Object {}}
    >
      Toast
    </text>
  </view>
</view>
`;
